packages/web-components/fast-components/src/switch/switch.styles.ts

```diff
@@ -57,214 +51,159 @@ export const switchStyles: (
              * Chromium likes to select label text or the default slot when
              * the checkbox is clicked. Maybe there is a better solution here?
              */ ""
-        } user-select: none;
+        }
+        user-select: none;
+        width: max-content;
     }
-
-    :host([disabled]) {
+    :host(.disabled) {
         opacity: ${disabledOpacity};
     }
-
-    :host([disabled]) .label,
-    :host([readonly]) .label,
-    :host([readonly]) .switch,
-    :host([disabled]) .switch {
+    :host(.disabled) .label,
+    :host(.readonly) .label,
+    :host(.disabled) .switch,
+    :host(.readonly) .switch,
+    :host(.disabled) .status-message,
+    :host(.readonly) .status-message {
         cursor: ${disabledCursor};
     }
-
     .switch {
         position: relative;
         outline: none;
         box-sizing: border-box;
-        width: calc(${heightNumber} * 1px);
-        height: calc((${heightNumber} / 2 + ${designUnit}) * 1px);
+        width: calc((${heightNumber} - (${designUnit} * 2 )) * 1px);
+        height: calc((${heightNumber} / 2) * 1px);
         background: ${neutralFillInputRest};
-        border-radius: calc(${controlCornerRadius} * 1px);
-        border: calc(${strokeWidth} * 1px) solid ${neutralStrokeRest};
+        border-radius: calc(${heightNumber} * 1px);
+        border: calc(${strokeWidth} * 1px) solid ${strokeControlStrongRest};
     }
-
-    .switch:hover {
-        background: ${neutralFillInputHover};
-        border-color: ${neutralStrokeHover};
-        cursor: pointer;
+    :host([aria-checked="true"]) .switch {
+        border-color: ${strokeControlStrongRest};
```

Comment:
For improved accessibility I updated the design to have the accent color in selected state for checkbox, radio, and switch.

Reply:
updated to add accent